lib/Doctrine/ORM/EntityRepository.php

```diff
@@ -221,21 +237,28 @@ public function __call($method, $arguments)
                 $method = 'findOneBy';
                 break;
 
+            case (0 === strpos($method, 'countBy')):
```

Comment:
You mean moving the logic after the `switch` block to a new private method (`resolveMagicCall()` or something similar)?